### ISSUE TITLE
Fedora: firewalld default settings block WireGuard

### ISSUE THREAD

**Issue by m0nhawk** [angristan/wireguard-install#54]
I have used the script on clean Fedora installation in Vultr, but no clients (macOS, mobile, Windows, multiple ISPs) is able to connect with the following client log:
```
2020-03-04 10:24:17.135947: [NET] peer(ZJdh…yJW4) - Sending handshake initiation
2020-03-04 10:24:22.445891: [NET] peer(ZJdh…yJW4) - Handshake did not complete after 5 seconds, retrying (try 2)
2020-03-04 10:24:22.446094: [NET] peer(ZJdh…yJW4) - Sending handshake initiation
```

<details>
<summary>Server config</summary>

```
[Interface]
Address = 10.66.66.1/24,fd42:42:42::1/64
ListenPort = 1194
PrivateKey = EPfU...MA0E=
PostUp = iptables -A FORWARD -i wg0 -j ACCEPT; iptables -t nat -A POSTROUTING -o ens3 -j MASQUERADE; ip6tables -A FORWARD -i wg0 -j ACCEPT; ip6tables -t nat -A POSTROUTING -o ens3 -j MASQUERADE
PostDown = iptables -D FORWARD -i wg0 -j ACCEPT; iptables -t nat -D POSTROUTING -o ens3 -j MASQUERADE; ip6tables -D FORWARD -i wg0 -j ACCEPT; ip6tables -t nat -D POSTROUTING -o ens3 -j MASQUERADE
[Peer]
PublicKey = asYF...fpWc=
AllowedIPs = 10.66.66.2/32,fd42:42:42::2/128
PresharedKey = wCTA...ajQI=
```
</details>

<details>
<summary>Client config</summary>

```
[Interface]
PrivateKey = oAUc...jUlA=
Address = 10.66.66.2/24,fd42:42:42::2/64
DNS = 176.103.130.130,176.103.130.131
[Peer]
PublicKey = ZJdh…yJW4=
Endpoint = <vultr>:1194
AllowedIPs = 0.0.0.0/0,::/0
PresharedKey = wCTA...ajQI=
```
</details>

<details>
<summary><code>sysctl -p</code></summary>

```
[root@vultr ~]# sysctl -p
...
net.ipv4.ip_forward = 1
net.ipv6.conf.all.forwarding = 1
```
</details>

---

Server does receive requests, but somehow it's being dropped (as seen from tcpdumps).

<details>
<summary><code>tcpdump -n -i ens3 "port 1194"</code></summary>

```
[root@vultr ~]# tcpdump -n -i ens3 "port 1194"
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on ens3, link-type EN10MB (Ethernet), capture size 262144 bytes
17:27:04.203131 IP 128.135.98.221.62114 > 80.240.22.24.openvpn: UDP, length 148
17:27:09.478064 IP 128.135.98.221.62114 > 80.240.22.24.openvpn: UDP, length 148
17:27:14.643509 IP 128.135.98.221.62114 > 80.240.22.24.openvpn: UDP, length 148
```
</details>

<details>
<summary><code>tcpdump -n -i wg0</code></summary>

```
[root@vultr ~]# tcpdump -n -i wg0
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on wg0, link-type RAW (Raw IP), capture size 262144 bytes
```
</details>

---

I tried to do manually based on [article](https://angristan.xyz/2019/01/how-to-setup-vpn-server-wireguard-nat-ipv6/) and this [manual](https://fedoramagazine.org/build-a-virtual-private-network-with-wireguard/), but without success.

**Comment by angristan** [comment]
Did you try with different Linux distributions?

**Comment by angristan** [comment]
It seems I can reproduce this.

**Comment by angristan** [comment]
Oh ok it's because firewalld is enabled by default.

**Comment by angristan** [comment]
See https://blog.airey.be/wireguard-setup.html for `firewall-cmd` commands

**Comment by m0nhawk** [comment]
@angristan Thank you for help.
Meanwhile I do manage to get it working with Ubuntu and Debian, so it's Fedora specific and the latest links helps.
```
PostUp     = firewall-cmd --add-port 1194/udp && firewall-cmd --add-rich-rule='rule family=ipv4 source address=10.66.66.0/24 masquerade'
PostDown   = firewall-cmd --remove-port 1194/udp && firewall-cmd --remove-rich-rule='rule family=ipv4 source address=10.66.66.0/24 masquerade'
```

I'm closing the issue, because it's not a bug of `wireguard-install`, but Fedora specific.

I'll make a PR with Fedora related changes, to make this part of project.

**Comment by angristan** [comment]
I'm keeping this open, since I would also prefer to have this included!

**Comment by m0nhawk** [comment]
Closed by #55
